File: Lista03_HerancaEPolimorfismo/Heranca/app/src/main/java/Lista03/Produto.java
package Lista03;

public class Produto{
    protected int codigo;
    protected double preco_unitario;
    protected String descricao;
    protected int quantidade_estoque;

    public Produto(int codigo, double preco_unitario, String descricao, int quantidade_estoque) {
        this.codigo = codigo;
        this.preco_unitario = preco_unitario;
        this.descricao = descricao;
        this.quantidade_estoque = quantidade_estoque;
    }

    public int getCodigo() {
        return codigo;
    }

    public double getPreco_unitario() {
        return preco_unitario;
    }

    public void setPreco_unitario(double preco_unitario) {
        this.preco_unitario = preco_unitario;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public int getQuantidade_estoque() {
        return quantidade_estoque;
    }

    public int retiraProduto(int n, Estoque estoque) {
        
        if (this.quantidade_estoque >= n){
            this.quantidade_estoque -= n;
            return n;
        }

        int q = this.quantidade_estoque;
        this.quantidade_estoque = 0;

        return q;
    }
    
    public boolean adicionarProduto(int n) {
        this.quantidade_estoque += n;
        return true;
    }

    public void imprimirProduto() {

        System.out.println("--------------------");
        System.out.println("Produto "+this.codigo+", "+this.descricao+", custo de R$"+this.preco_unitario+", quantidade "+this.quantidade_estoque);
        System.out.println("--------------------");

        return;
    }

}

File: Lista11_Python/Ex04_Heranca/produto.py
from abc import ABC, abstractmethod

class Produto(ABC):

    def __init__(self, codigo, precoUnitario, descricao, qtdEstoque):
        self._codigo = codigo
        self._precoUnitario = float(precoUnitario)
        self._descricao = descricao
        self._qtdEstoque = qtdEstoque

    def __str__(self):
        string = "Produto: "+str(self._codigo)+", "+self._descricao+", custo de R$"+str(self._precoUnitario)+", quantidade "+str(self._qtdEstoque)
        return string

    def getPreco(self):
        return self.preco

    def setPreco(self, preco):
        self._precoUnitario = preco

    def getDescricao(self):
        return self._descricao

    def setDescricao(self, descricao):
        self._descricao = descricao

    def getCodigo(self):
        return self._codigo

    def getEstoque(self):
        return self._qtdEstoque

    def retiraEstoque(self, qtd_retirada):
        self._qtdEstoque -= qtd_retirada
        if self._qtdEstoque < 0:
            retorno = qtd_retirada + self._qtdEstoque
            self._qtdEstoque = 0
            return (retorno)
        else:
            return qtd_retirada

    def acrescentaEstoque(self, qtd_acrescentada):
        self._qtdEstoque += qtd_acrescentada
